File: 프로그래머스/unrated/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.java
import java.util.ArrayList;
import java.util.Arrays;

public class Solution {
    public int[] solution(String my_string) {
        ArrayList<Integer> list = new ArrayList<>();
        
        for (int i = 0; i < my_string.length(); i++) {
            if (Character.isDigit(my_string.charAt(i))){
                list.add(Character.getNumericValue(my_string.charAt(i)));
            }
        }
        
        int[] answer = new int[list.size()];
        
        for (int i = 0; i < answer.length; i++) {
            answer[i] = list.get(i); 
        }
        
        Arrays.sort(answer);
        
        return answer;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        String my_string = "hi12392";
        int[] answer = sol.solution(my_string);
        System.out.println("answer = " + Arrays.toString(answer));
    }
}

File: JavaScript/프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.js
function solution(my_string) {
    const answer = [];
    
    my_string.split("").forEach(element => {
        if (/\d/.test(element)) {  // 정규식으로 숫자 판별
            answer.push(parseInt(element, 10)); // 숫자를 정수로 변환하여 배열에 추가
        }
    });

    return answer.sort();
}